docs/loading/Flink-connector-starrocks.md

```diff
@@ -1,173 +1,647 @@
+
+- If data loading pauses, you can try to increase the memory of the Flink task.
 
 - If the preceding code runs as expected and StarRocks can receive data, but the data loading fails, check whether your machine can access the HTTP port of the backends (BEs) in your StarRocks cluster. If you can successfully ping the HTTP port returned by the execution of the SHOW BACKENDS command in your StarRocks cluster, your machine can access the HTTP port of the BEs in your StarRocks cluster. For example, a machine has a public IP address and a private IP address, the HTTP ports of frontends (FEs) and BEs can be accessed through the public IP address of the FEs and BEs, the IP address that is bounded with your StarRocks cluster is the private IP address, and the value of `loadurl` for the Flink task is the HTTP port of the public IP address of the FEs. The FEs forwards the data loading task to the private IP address of the BEs. In this example, if the machine cannot ping the private IP address of the BEs, the data loading fails.
+
+- If you specify the value as exactly-once, `sink.buffer-flush.max-bytes`, `sink.buffer-flush.max-bytes`, and `sink.buffer-flush.interval-ms` are invalid.
+
+## Examples
+
+The following examples show how to use the Flink connector to load data into a StarRocks table with Flink SQL or Flink DataStream.
+
+### Preparations
+
+#### Create a StarRocks table
+
+Create a database `test` and create a Primary Key table `score_board`.
+
+```sql
+CREATE DATABASE `test`;
+
+CREATE TABLE `test`.`score_board`
+(
+    `id` int(11) NOT NULL COMMENT "",
+    `name` varchar(65533) NULL DEFAULT "" COMMENT "",
+    `score` int(11) NOT NULL DEFAULT "0" COMMENT ""
+)
+ENGINE=OLAP
+PRIMARY KEY(`id`)
+COMMENT "OLAP"
+DISTRIBUTED BY HASH(`id`);
+```
+
+#### Set up Flink environment
+
+- Download Flink binary [Flink 1.15.2](https://archive.apache.org/dist/flink/flink-1.15.2/flink-1.15.2-bin-scala_2.12.tgz), and unzip it to directory `flink-1.15.2`.
+- Download [connector 1.2.7](https://repo1.maven.org/maven2/com/starrocks/flink-connector-starrocks/1.2.7_flink-1.15/flink-connector-starrocks-1.2.7_flink-1.15.jar), and put it into the directory `flink-1.15.2/lib`.
+- Run the following commands to start a Flink cluster:
+
+    ```shell
+    cd flink-1.15.2
+    ./bin/start-cluster.sh
+    ```
+
+### Run with Flink SQL
+
+- Run the following command to start a Flink SQL client.
+
+    ```shell
+    ./bin/sql-client.sh
+    ```
+
+- Create a Flink table `score_board`, and insert values into the table via Flink SQL Client.
+Note you must define the primary key in the Flink DDL if load to a StarRocks primary key table. It's optional for other types of StarRocks table.
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/b558e3c6-88c0-4405-adb8-65beb4b8349a)
-> if you want to load data into a Primary Key table of StarRocks